core/src/main/scala/cats/data/StateT.scala

```diff
@@ -133,6 +133,12 @@ object StateT extends StateTInstances {
 
   def pure[F[_], S, A](a: A)(implicit F: Applicative[F]): StateT[F, S, A] =
     StateT(s => F.pure((s, a)))
+
+  def lift[F[_], S, A](fa: F[A])(implicit F: Applicative[F]): StateT[F, S, A] =
+    StateT(s => F.map(fa)(a => (s, a)))
+
+  def inspect[F[_], S, A](f: S => F[A])(implicit F: Applicative[F]): StateT[F, S, A] =
+    StateT(s => F.map(f(s))(a => (s, a)))
```

Comment:
Should we have `inspect` take `S => A` and create a separate `inspectF` for `S => F[A]`? It seems like the simple `S => A` case might be pretty common, so it might be nice to make it concise.